src/utils/thread_helper.cpp

```diff
@@ -944,6 +944,44 @@ void ThreadHelper::DetachGracefullyCallback(void)
 
 #if OTBR_ENABLE_TELEMETRY_DATA_API
 #if OTBR_ENABLE_BORDER_ROUTING
+void ThreadHelper::RetrieveInfraLinkInfo(threadnetwork::TelemetryData::InfraLinkInfo *aInfraLinkInfo)
+{
+    {
+        otSysInfraNetIfAddressCounters addressCounters;
+        uint32_t                       ifrFlags = otSysGetInfraNetifFlags();
+
+        otSysCountInfraNetifAddresses(&addressCounters);
+
+        aInfraLinkInfo->set_name(otSysGetInfraNetifName());
+        aInfraLinkInfo->set_is_up((ifrFlags & IFF_UP) != 0);
```

Comment:
Since you are uploading the ifrFlags contents, how about directly upload ifrFlags itself, if there is no sensitive information inside? This will save the upload payload size.

Reply:
thanks @morningboata , these block is just the refactor, I think it's better to keep as it original is?  cc. @superwhd

Reply:
I don't think ifr_flags' integer values are the same across platforms. Hence we have to extract every flag using the platform-specific enums before post-processing.